File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/equals/EqualsFloatingPoint.java
import java.util.Objects;

public class J {
    public static void main(String[] args) {
        Float posZero = 0.0f;
        Double negZero = -0.0;
        System.out.println(posZero.equals(negZero));
        System.out.println(Objects.equals(posZero, negZero));
        System.out.println(java.util.Objects.equals(posZero, negZero));
        System.out.println(posZero.equals(0));
        System.out.println(Objects.equals(0, negZero));
        System.out.println(Objects.equals(0, 0));
    }
}

File: plugins/kotlin/j2k/new/tests/testData/newJ2k/equals/EqualsFloatingPoint.kt
import java.util.Objects

object J {
    @JvmStatic
    fun main(args: Array<String>) {
        val posZero = 0.0f
        val negZero = -0.0
        println(posZero.equals(negZero))
        println(Objects.equals(posZero, negZero))
        println(Objects.equals(posZero, negZero))
        println(posZero.equals(0))
        println(Objects.equals(0, negZero))
        println(0 == 0)
    }
}
